src/vm_cpu.rs

```diff
@@ -361,13 +407,15 @@ impl VmPmuState {
                         let initial_value = c.value;
                         if matches!(state, Started(_)) {
                             let start_flags = PmuCounterStartFlags::default().set_init_value();
-                            // TODO: Use result for counter virtualization
-                            let _result = sbi::api::pmu::start_counters(
+                            let result = sbi::api::pmu::start_counters(
```

Comment:
What if this fails? Should we poison the counter?

Reply:
Yup...it was being poisoned before, but got lost when the method was extracted. Changed to poison on failure.